### PR TITLE
Pre madrid implementations

### PR DESCRIPTION
**Pull request by Omair95**

### CODE REVIEW

File: src/js/browser-plugin.js [hunk 1]
```diff
@@ -69,66 +56,81 @@ const start = () => {
...
+
   tweetData = tweetInfo;
+  dom = tweetInfo.domObject;
   if (tweetInfo.links.length > 0) {
-    const score = {};
 
-    if (usersCache.hasOwnProperty(tweetInfo.username)) {
-      score.username = tweetInfo.username;
-      score.misinformationScore = usersCache[tweetInfo.username];
+    if (usersCache[tweetInfo.id] == null) usersCache[tweetInfo.id] = false;
```
Comment (davidsalvador) [inline src/js/browser-plugin.js]: Isn't the linter (JSHint) marking this as bad code style?

File: src/js/browser-plugin.js [hunk 2]
```diff
@@ -138,24 +140,128 @@ const newFacebookPostCallback = (post) => {
...
+    if (x == 0) {
+      node.append(addCoinformLogo(tweet));
+    }
   }
-
 };
 
+const addCoinformLogo = (tweet) => {
+  var x = document.createElement("img");
+  x.setAttribute("src", 'https://en.gravatar.com/userimage/157153645/613b113e876d7942df3056fe07436977?size=200');
```
Comment (davidsalvador) [inline src/js/browser-plugin.js]: Could we include this image in the plugin resources? Ideally we shouldn't have to make extra HTTP requests for this kind of stuff.

File: src/js/browser-plugin.js [hunk 3]
```diff
@@ -229,43 +327,53 @@ const createWhyButton = (publicationName, addPaddingTop = false) => {
...
+          returned = returned.toString();
+          var array = returned.split(',');
+          var url = array[0], comment = array[1];
 
-          client.postTwitterEvaluate(tweetData.id, evaluation)
-          .then(res => {
-            
-            console.log('Query ID = ' + JSON.stringify(res));
+          if (url.localeCompare('') === 0) {
+            alert('You need to insert an URL!');
```
Comment (davidsalvador) [inline src/js/browser-plugin.js]: I would suggest saying "Please provide a valid URL" instead.

File: src/js/browser-plugin.js [hunk 4]
```diff
@@ -229,43 +327,53 @@ const createWhyButton = (publicationName, addPaddingTop = false) => {
...
+          }
+          else if (comment.localeCompare('') === 0) {
+            alert('You need to insert a comment!');
```
Comment (davidsalvador) [inline src/js/browser-plugin.js]: I would suggest saying "Please provide some comment" instead.

File: src/js/browser-plugin.js [hunk 5]
```diff
@@ -229,43 +327,53 @@ const createWhyButton = (publicationName, addPaddingTop = false) => {
...
+              
+              console.log('Query ID = ' + JSON.stringify(res));
+              
+            })
+            .catch(err => console.log(err));
+
+            resolve();
+          }  
+        })
+      } else if (result.dismiss == 'cancel') {
```
Comment (davidsalvador) [inline src/js/browser-plugin.js]: Use `===` instead of `==` when comparing in JavaScript.

Comment (Omair95) [reply src/js/browser-plugin.js]: Changed

File: src/js/browser-plugin.js (hunk 4 above)
Comment (Omair95) [reply src/js/browser-plugin.js]: Changed.

File: src/js/browser-plugin.js (hunk 3 above)
Comment (Omair95) [reply src/js/browser-plugin.js]: Changed.

File: src/js/browser-plugin.js (hunk 2 above)
Comment (Omair95) [reply src/js/browser-plugin.js]: I was getting a Cross-Origin Read Blocking (CORB) error while trying to access my local logo image thats why as a working-solution I implemented this.

File: src/js/browser-plugin.js (hunk 1 above)
Comment (Omair95) [reply src/js/browser-plugin.js]: Yes, i forgot to pass the hinter (other error were showing too).

File: src/js/browser-plugin.js (hunk 2 above)
Comment (Omair95) [reply src/js/browser-plugin.js]: I just reimplemented it using the local resource icons.

**Pull request merged**